lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1069,15 +1078,69 @@ SPIRVValue *SPIRVModuleImpl::addNullConstant(SPIRVType *Ty) {
 
 SPIRVValue *SPIRVModuleImpl::addCompositeConstant(
     SPIRVType *Ty, const std::vector<SPIRVValue *> &Elements) {
+  SPIRVConstantComposite *Res;
+  const size_t MaxNumElements = 65535 - 3;
+  const size_t NumElements = Elements.size();
+  if (NumElements > MaxNumElements &&
+      isAllowedToUseExtension(ExtensionID::SPV_INTEL_long_constant_composite)) {
```

Comment:
What about error handling when extension is disabled?

Reply:
Same here, error will be emitted by `validate` functionality of corresponding class.

Reply:
Maybe it would be helpful to add comment on that before line 1103.